hexa/datasets/queue.py

```diff
@@ -0,0 +1,105 @@
+import json
+from logging import getLogger
+
+import pandas as pd
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from django.db import DatabaseError, IntegrityError
+from dpq.queue import AtLeastOnceQueue
+
+from hexa.datasets.api import generate_download_url
+from hexa.datasets.models import (
+    DatasetFileMetadata,
+    DatasetFileMetadataJob,
+    DatasetVersionFile,
+)
+
+logger = getLogger(__name__)
+
+
+def download_file_as_dataframe(
+    dataset_version_file: DatasetVersionFile,
+) -> pd.DataFrame:
+    filename = dataset_version_file.filename
+    file_format = filename.split(".")[-1]
```

Comment:
It would make sense to use this https://docs.python.org/3/library/mimetypes.html